oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/dao/BanyanDBBatchDAO.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.storage.plugin.banyandb.dao;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.skywalking.apm.commons.datacarrier.DataCarrier;
+import org.apache.skywalking.apm.commons.datacarrier.consumer.BulkConsumePool;
+import org.apache.skywalking.apm.commons.datacarrier.consumer.ConsumerPoolFactory;
+import org.apache.skywalking.apm.commons.datacarrier.consumer.IConsumer;
+import org.apache.skywalking.oap.server.core.UnexpectedException;
+import org.apache.skywalking.oap.server.library.client.jdbc.hikaricp.JDBCHikariCPClient;
+import org.apache.skywalking.oap.server.library.client.request.InsertRequest;
+import org.apache.skywalking.oap.server.library.client.request.PrepareRequest;
+import org.apache.skywalking.oap.server.storage.plugin.banyandb.BanyanDBStorageConfig;
+import org.apache.skywalking.oap.server.storage.plugin.banyandb.client.BanyanDBGrpcClient;
+import org.apache.skywalking.oap.server.storage.plugin.banyandb.client.BanyanDBInsertRequest;
+import org.apache.skywalking.oap.server.storage.plugin.jdbc.SQLExecutor;
+import org.apache.skywalking.oap.server.storage.plugin.jdbc.h2.dao.H2BatchDAO;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class BanyanDBBatchDAO extends H2BatchDAO {
+    private final DataCarrier<PrepareRequest> dataCarrier;
```

Comment:
I don't think you need a new dataCarrier. You have a streaming client. The reason H2 has to add one like this, because we don't want to open transaction for every record, but you don't face the same issue here.

Reply:
You are just using H2Dao to cover the blank APIs, so, it wouldn't concern about the performance. Let's remove the queue until we have to add it back. Take a look at the ElasticSearch DAOs implementation. Those are really product level, which process 100+ million traces/logs per seconds.

Reply:
Make sense to me. I was confused about this Batch API...

Reply:
Batch API means streaming write, which is why we designed in proto level.